src/main.rs

```diff
@@ -197,51 +197,57 @@ fn decode(input_path: &Path, restored_path: &Path) -> Result<(), parser::Restore
 }
 
 fn main() -> Result<(), parser::RestoreError> {
-    let matches = App::new("qrxfil")
-        .version("0.1")
-        .about("Transfer/backup files as a sequence of QR codes")
-        .author("Jb DOYON") // And authors
-        .subcommand(
-            SubCommand::with_name("exfil")
-                .about("Generates QR code sequence from file")
-                .arg(
-                    Arg::with_name("input") // And their own arguments
-                        .help("The input file to split into QR codes")
-                        .index(1)
-                        .required(true),
-                )
-                .arg(
-                    Arg::with_name("output_folder")
-                        .help("The output folder to generate codes into")
-                        .index(2)
-                        .required(true),
-                ),
-        )
-	.subcommand(
-	                SubCommand::with_name("restore")
-                .about("Decodes encoded strings back into file")
-                .arg(
-                    Arg::with_name("encoded_input") // And their own arguments
-                        .help("The input file with newline-delimited QR strings")
-                        .index(1)
-                        .required(true),
-                )
-                .arg(
-                    Arg::with_name("output_file")
-                        .help("The output file to restore into")
-                        .index(2)
-                        .required(true),
-                ),
-
-	)
-        .get_matches();
-
-    // // You can check if a subcommand was used like normal
-    // let matches_exfil = match matches.subcommand_matches("exfil") {
-    //     Some(i) => i,
-    //     None => panic!("Exfil alone is implemented"),
-    // };
+    let matches = get_args();
+    let result = run(&matches);
+    if let Err(ref e) = result {
+        eprintln!("{}", e);
+    }
+
+    result
+}
+
+fn get_args() -> ArgMatches<'static> {
+    App::new("qrxfil")
+    .version("0.1")
+    .about("Transfer/backup files as a sequence of QR codes")
+    .author("Jb DOYON") // And authors
+    .subcommand(
+        SubCommand::with_name("exfil")
+            .about("Generates QR code sequence from file")
+            .arg(
+                Arg::with_name("input") // And their own arguments
+                    .help("The input file to split into QR codes")
+                    .index(1)
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("output_folder")
+                    .help("The output folder to generate codes into")
+                    .index(2)
+                    .required(true),
+            ),
+    )
+.subcommand(
```

Comment:
weird indent

Reply:
is it? i just ran `cargo fmt`.

speaking of, your project could do with a `.rustfmt.toml` file

Reply:
yeah rustfmt isn't doing anything for some reason. suspect because there are errors in the codebase? seems to be silently failing... weird

Reply:
seems to be a manifestation of this issue - https://github.com/rust-lang/rustfmt/issues/3863#issuecomment-699768919

i think a workaround might be to add `.rustfmt.toml` file with the `format_strings = true` option enabled

Reply:
fixed